### PR TITLE
initial re-port of requirements to get a build to work on windows

### PR DESCRIPTION
**Pull request by steffengy**
This commit does not include the collected C-headers yet, to keep it easily diffable.

### CODE REVIEW

**Comment by cznic** [comment]
Thanks for the PR. I have a 4-day weekend, but unfortunately 3 of them we're mostly away from home. Not exactly sure when I can get to the review.

File: internal/c99/ast2.go [hunk 1]
```diff
@@ -195,6 +195,14 @@ func (d *DeclarationSpecifier) typ(ctx *context) Type {
 		return UShort
 	case d.is(TypeSpecifierShort, TypeSpecifierSigned):
 		return Short
+	// TODO: not sure if this is right
```
Comment (cznic) [inline internal/c99/ast2.go]: It does not look correct to me. It probably indicates a bug in the frontend. Can you please provide a small example that triggers this case? You can put it right into c99/testdata/bug/nn.c, thanks.

File: internal/c99/headers/windows_amd64/builtin.h [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+#define BYTE unsigned char
+#define __int16 short
+#define __int32 int
+#define __int64 long long
+// disable SSE and other instrisics we dont support
+#define _X86INTRIN_H_INCLUDED 1
+#undef __SSE__
+#undef __SSE2__
+#undef __MMX__
+// bypass bug that fails for: #define MICROSOFT_WINDOWS_WINBASE_H_DEFINE_INTERLOCKED_CPLUSPLUS_OVERLOADS (_WIN32_WINNT >= 0x0502 || !defined (_WINBASE_))
```
Comment (cznic) [inline internal/c99/headers/windows_amd64/builtin.h]: Assuming the bug is probably in c99/cpp.go: Can you please add a small repro case in either c99/testdata/cpp-expand or c99/testdata/bug/nn.c if the bug is elsewhere?

File: internal/c99/headers/windows_amd64/builtin.h [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+#define __unaligned
+#define __inline__ inline
+#define __forceinline inline __attribute__((__always_inline__,__gnu_inline__))
+
+// TODO: __builtin_fprintf(stderr, "%s:%i.%s STUB %s called!\n", __FILE__, __LINE__, __func__, #x), __builtin_abort(), 0)
+#define abort_stubbed(x) 0
+
+// maybe: `#define __CRT__NO_INLINE` to remove some unnecessary stuff
+// Implementing __readgsqword as macro also prevents an anonymous union access 
+// (which isn't supported in C99 actually WTF)
```
Comment (cznic) [inline internal/c99/headers/windows_amd64/builtin.h]: I don't mind the `WTF`, but please consider some other form for others out there ;-)

File: internal/c99/headers/windows_amd64/builtin.h [hunk 4]
```diff
@@ -0,0 +1,137 @@
...
+#define __builtin_types_compatible_p(type1, type2) __builtin_types_compatible__((type1){}, (type2){})
+#define __complex__ _Complex
+#define __const const
+#define __extension__
+#define __inline inline
+#define __restrict restrict
+#define __roundup(n, mod) ((n + mod - 1) & ~(mod - 1))
+#define __volatile volatile
+
+typedef void *__FILE_TYPE__;
```
Comment (cznic) [inline internal/c99/headers/windows_amd64/builtin.h]: Note that the linux variant now uses whatever the system headers define `FILE` to be. I don't think it's a problem ATM, though.

File: internal/c99/ast2.go (hunk 1 above)
Comment (steffengy) [reply internal/c99/ast2.go]: What's the easiest way to track this down to the original file location?
Doing something like fmt.Println(d.TypeSpecifiers[0].Pos()) only gives me the position of the token and not the file.

Comment (cznic) [reply internal/c99/ast2.go]: I hope `ctx.position(d)` should provide that.

Comment (cznic) [reply internal/c99/ast2.go]: No, `DeclarationSpecifier` is not a `Node`. But any non nil item within the `*FunctionSpecifier`, ... slices is a `Node`. I should add the `.Pos()` method to `DeclarationSpecifier`.

Comment (steffengy) [reply internal/c99/ast2.go]: It seems to work with ctx.position(d.TypeSpecifiers[0]))

Comment (steffengy) [reply internal/c99/ast2.go]: Reproduction in https://github.com/cznic/sqlite2go/pull/7/commits/27c997917dab8ef7e3a765bb6a19ec54782afea7
(testdata/bugs/20.c)

File: internal/c99/headers/windows_amd64/builtin.h (hunk 2 above)
Comment (steffengy) [reply internal/c99/headers/windows_amd64/builtin.h]: reproduction in https://github.com/cznic/sqlite2go/pull/7/commits/009e2de130daf943a72d3af25a131bf0530731f6

**Pull request merged**